central/role/service/service_impl_test.go

```diff
@@ -454,3 +495,390 @@ func TestEffectiveAccessScopeForSimpleAccessScope(t *testing.T) {
+		s.Require().NoError(err)
+		namespaceStore, err := namespaceDataStore.GetTestRocksBleveDataStore(s.T(), s.rocksEngine, s.bleveIndex, dacky, keyFence)
+		s.Require().NoError(err)
+
+		s.service = &serviceImpl{
+			roleDataStore:      roleStore,
+			clusterDataStore:   clusterStore,
+			namespaceDataStore: namespaceStore,
+		}
+	}
+}
+
+func (s *serviceImplTestSuite) TearDownSuite() {
+	if env.PostgresDatastoreEnabled.BooleanSetting() {
+		s.postgres.Teardown(s.T())
+	} else {
+		s.Require().NoError(s.boltEngine.Close())
+		s.Require().NoError(rocksdb.CloseAndRemove(s.rocksEngine))
+		s.Require().NoError(s.bleveIndex.Close())
+	}
+}
+
+func (s *serviceImplTestSuite) SetupTest() {
+	s.storedClusterIDs = make([]string, 0)
+	s.storedNamespaceIDs = make([]string, 0)
+}
+
+func (s *serviceImplTestSuite) TearDownTest() {
+	writeCtx := sac.WithAllAccess(context.Background())
+	doneSignal := concurrency.NewSignal()
+	for _, clusterID := range s.storedClusterIDs {
+		s.NoError(s.service.clusterDataStore.RemoveCluster(writeCtx, clusterID, &doneSignal))
+	}
+	<-doneSignal.Done()
+	s.storedClusterIDs = s.storedClusterIDs[:0]
+	for _, namespaceID := range s.storedNamespaceIDs {
+		s.NoError(s.service.namespaceDataStore.RemoveNamespace(writeCtx, namespaceID))
+	}
+}
+
+const (
+	namespaceUUIDNamespace = "namespace"
+
+	clusterPermission             = "Cluster"
+	deploymentPermission          = "Deployment"
+	deploymentExtensionPermission = "DeploymentExtension"
+	integrationPermission         = "Integration"
+	namespacePermission           = "Namespace"
+	networkGraphPermission        = "NetworkGraph"
+	nodePermission                = "Node"
+	rolePermission                = "Role"
+)
+
+func getTestResourceScopeSingleNamespace(_ *testing.T, clusterID string, namespace string) *sac.TestResourceScope {
+	return &sac.TestResourceScope{
+		Clusters: map[string]*sac.TestClusterScope{
+			clusterID: {
+				Namespaces: []string{namespace},
+				Included:   false,
+			},
+		},
+		Included: false,
+	}
+}
+
+func getNamespaceID(namespaceName string) string {
+	return uuid.NewV5FromNonUUIDs(namespaceUUIDNamespace, namespaceName).String()
+}
+
+func (s *serviceImplTestSuite) TestGetClustersForPermissions() {
+	writeCtx := sac.WithAllAccess(context.Background())
+	clusterNameToIDMap := make(map[string]string, 0)
+	for _, cluster := range clusters {
+		clusterToAdd := cluster.Clone()
+		clusterToAdd.Id = ""
+		id, err := s.service.clusterDataStore.AddCluster(writeCtx, clusterToAdd)
+		s.Require().NoError(err)
+		clusterNameToIDMap[cluster.GetName()] = id
+		s.storedClusterIDs = append(s.storedClusterIDs, id)
+	}
+	for _, namespace := range namespaces {
+		ns := namespace.Clone()
+		ns.Id = getNamespaceID(ns.GetName())
+		s.Require().NoError(s.service.namespaceDataStore.AddNamespace(writeCtx, ns))
+		s.storedNamespaceIDs = append(s.storedNamespaceIDs, ns.GetId())
+	}
```

Comment:
Shouldn't we do this within the `TestSetup` function and make `clusterNameToIDMap` a property of the suite?